Source/WebCore/platform/graphics/MediaSourcePrivate.cpp

```diff
@@ -189,6 +189,13 @@ void MediaSourcePrivate::ensureOnDispatcher(Function<void()>&& function) const
     m_dispatcher->dispatch(WTFMove(function));
 }
 
+MediaTime MediaSourcePrivate::currentMediaTime() const
+{
+    if (RefPtr player = this->player())
+        return player->apparentMediaTime();
+    return MediaTime::invalidTime();
```

Comment:
```suggestion
    RefPtr player = this->player();
    if (!player)
        return MediaTime::invalidTime();

    auto pendingSeekTime = player->pendingSeekTime();
    return pendingSeekTime.isValid() ? pendingSeekTime : player-> currentMediaTime();
```

Reply:
issue with this is that it is possible for pendingSeekTime to change before we call currentMediaTime.

Likelihood is going to be very low, but it is necessary I believe for having synchronise read of both pendingSeekTime and currentMediaTime.